### ISSUE TITLE
STDIN an local Setting not working

### ISSUE THREAD

**Issue by sv-dl** [cmhughes/latexindent.pl#189]
Hallo,

I'm trying to use latexindent as a filter and find out that this will not work with using local setting files like
`cat doc.tex | latexindent.pl -l myConf.yaml`
Trys to fomat `myConf.yaml` instead of `doc.tex`

I've found the reason in `latexindent.pl` well documente

```
# check local settings doesn't interfer with reading the file;
# this can happen if the script is called as follows:
#
#       latexindent.pl -l myfile.tex
#
# in which case, the GetOptions routine mistakes myfile.tex
# as the optional parameter to the l flag.
#
# In such circumstances, we correct the mistake by assuming that 
# the only argument is the file to be indented, and place it in @ARGV
if($switches{readLocalSettings} and scalar(@ARGV) < 1) {
    push(@ARGV,$switches{readLocalSettings});
    $switches{readLocalSettings} = ''; 
}
```

I'm not really sure what's the right way to handle the problem. Maybe there should be a extra option to tell latexindent.pl that I really want to read from STDIN or test if there is something to read from STDIN or check if the file after -l has an extension and is a yaml-file.

Thanks for this nice tool
regards
  sv-dl

**Comment by sv-dl** [comment]
a work around is specifying `-` as document like

`cat doc.tex | latexindent.pl -l myConf.yaml - `

**Comment by cmhughes** [comment]
Thanks very much for this, and also for the work around -- I very much appreciate you debugging this! I think that your work around is the appropriate way to handle this.

I admit that I hadn't come across this issue before, as I hadn't thought of it! I've updated the documentation to detail this, as it wasn't explicitly detailed previously. 

Thank you!

**Comment by cmhughes** [comment]
Implemented in `Version 3.8`, https://github.com/cmhughes/latexindent.pl/pull/190
